From 850b23f6ba9affb949918a8f670ef0a1606c42a2 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Thu, 18 May 2017 10:51:27 +0800
Subject: [PATCH] ARM: rockchip: Add workaround for unknown write of
 thread_info

We see the cpu of thread_info was changed by unknown reason sometimes,
which cause rq spinlock deadlock while schedule. Until we found the
root cause, we have to add this workaround.

Change-Id: Ib943ccb47a57fe4b267a1da853366363cd7f1f52
Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm/include/asm/thread_info.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/include/asm/thread_info.h b/arch/arm/include/asm/thread_info.h
index 776757d1604a..af4a57fdc816 100644
--- a/arch/arm/include/asm/thread_info.h
+++ b/arch/arm/include/asm/thread_info.h
@@ -47,6 +47,13 @@ struct cpu_context_save {
  * __switch_to() assumes cpu_context follows immediately after cpu_domain.
  */
 struct thread_info {
+#ifdef CONFIG_ARCH_ROCKCHIP
+	/*
+	 * FIXME: prevent unknown write of cpu
+	 * we reservee 64 bytes to make sure cacheline aligned
+	 */
+	__u8			reserved[64];
+#endif
 	unsigned long		flags;		/* low level flags */
 	int			preempt_count;	/* 0 => preemptable, <0 => bug */
 	mm_segment_t		addr_limit;	/* address limit */
-- 
2.35.3

